image/image.go

```diff
@@ -115,6 +122,10 @@ func Unpack(tarFile, dest, ref string) error {
 }
 
 func unpack(w walker, dest, refName string) error {
+	if err := refsNameRegexpCheck(refName); err != nil {
```

Comment:
I'm not convinced that we want to be doing validation as part of unpacking (folks who want it can call the validation code *before* calling `unpack`).  But if that is not convincing, we can always add this for now, and remove it if/when someone gets around to filing a PR to pull the existing validation logic out of the unpacking code.